app/config.go

```diff
@@ -27,3 +57,33 @@ var evmosActivators = map[string]func(*vm.JumpTable){
 	"evmos_1": eips.Enable0001,
 	"evmos_2": eips.Enable0002,
 }
+
+// setBaseDenomWithChainID registers the display denom and base denom and sets the
+// base denom for the chain. The function registers different values based on
+// the chainID to allow different configurations in mainnet and testnet.
+func setBaseDenomWithChainID(chainID string) {
+	if utils.IsTestnet(chainID) {
+		if err := sdk.RegisterDenom(types.DisplayDenomTestnet, math.LegacyOneDec()); err != nil {
+			panic(err)
+		}
+		if err := sdk.RegisterDenom(types.BaseDenomTestnet, math.LegacyNewDecWithPrec(1, types.BaseDenomUnit)); err != nil {
+			panic(err)
+		}
+		if err := sdk.SetBaseDenom(types.BaseDenomTestnet); err != nil {
+			panic("can't set base denom")
+		}
+		return
+	}
+
+	if utils.IsMainnet(chainID) {
+		sdk.RegisterDenom(types.DisplayDenom, math.LegacyOneDec()) //nolint:errcheck
```

Comment:
this error is not checked

Reply:
The error is not checked if because if it was already registered, it's technically not an error. But we shoud check for the error value to ensure that is the already registered and not something else